src/ConfigFetcher.ts

```diff
@@ -1,49 +1,78 @@
-import type { IConfigFetcher, IFetchResponse, OptionsBase } from "configcat-common";
+import type {
+  AxiosError,
+  AxiosRequestConfig,
+  AxiosRequestHeaders,
+  AxiosResponse,
+} from "axios";
+import axios from "axios";
+import type {
+  IConfigFetcher,
+  IFetchResponse,
+  OptionsBase,
+} from "configcat-common";
 import { FetchError } from "configcat-common";
 
 export class HttpConfigFetcher implements IConfigFetcher {
-  private handleStateChange(httpRequest: XMLHttpRequest, resolve: (value: IFetchResponse) => void, reject: (reason?: any) => void) {
-    try {
-      if (httpRequest.readyState === 4) {
-        const { status: statusCode, statusText: reasonPhrase } = httpRequest;
+  async fetchLogic(
+    options: OptionsBase,
+    lastEtag: string | null
+  ): Promise<IFetchResponse> {
+    // If we are not running in browser set the If-None-Match header.
+    const headers: AxiosRequestHeaders | { "If-None-Match": string } =
```

Comment:
You set the ETag header (because of server-side mode) manually, do you know the content negotiation (Accept-Encoding: gzip) is set automatically when the usage is server-side? Unfortunately, I have not found any documentation on how Axios manages the content negotiation on the server side, I see that it can decompress the content automatically (decompress: true). So my question is should we set the `Accept-Encoding` header manually or not? If it is you should add this header too to the list: `Accept-Encoding: gzip`